src/main/java/org/openstreetmap/atlas/geography/atlas/pbf/OsmPbfProcessor.java

```diff
@@ -904,7 +927,7 @@ else if (entity instanceof Relation)
 
     private void tagOutsideBoundaryNodes(final Node node)
     {
-        if (this.nodeIdentifiersAtNetworkBoundary.contains(node.getId()))
+        if (this.nodeIdentifiersAtNetworkBoundary.contains(node.getId()) && isOutsideBoundary(node))
```

Comment:
what is the expectation for a node that falls directly on a boundary during country slicing?  Will it get a country code for the country it is on the boundary of?  Will nearest neighbor be false?  If that's the case, will a SyntheticBoundaryNodeTag get applied since the condition is that the node must be outside the boundary?

Reply:
For a `Node` directly on the boundary, it will get a country code for all countries it is on the boundary of. So it will have 2 or more country codes. Nearest neighbor will be `false` and `SyntheticBoundaryNodeTag` will be applied [here](https://github.com/osmlab/atlas/blob/8d6defc17ae02e318fe7c68a4209af5d197c817e/src/main/java/org/openstreetmap/atlas/geography/atlas/pbf/slicing/CountrySlicingProcessor.java#L940) or [here](https://github.com/osmlab/atlas/blob/8d6defc17ae02e318fe7c68a4209af5d197c817e/src/main/java/org/openstreetmap/atlas/geography/atlas/pbf/slicing/CountrySlicingProcessor.java#L969), depending on the case (if it is an existing node or a new node created by country slicing a way crossing the boundary).

Reply:
Ah, nice! Thanks @MikeGost .

Reply:
Forgot to mention, the first case that's linked above, where the existing OSM Node is on the boundary, is not handled in today's flow (see `if false` statement). It is handled in the Raw Atlas refactor [flow](https://github.com/osmlab/atlas/blob/a46f9baf6acd0240731468b89e2a87bb5925486d/src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicer.java#L237).